include/Symmetry.hpp

```diff
@@ -145,6 +160,56 @@ namespace cytnx {
   };
   ///@endcond
 
+  ///@cond
+  class FermionParitySymmetry : public Symmetry_base {
+   public:
+    FermionParitySymmetry() {
+      this->stype_id = SymType.fPar;
+      this->n = -2;
+    };
+    boost::intrusive_ptr<Symmetry_base> clone() {
+      boost::intrusive_ptr<Symmetry_base> out(new FermionParitySymmetry());
+      return out;
+    }
+    bool check_qnum(const cytnx_int64 &in_qnum);
+    bool check_qnums(const std::vector<cytnx_int64> &in_qnums);
+    void combine_rule_(std::vector<cytnx_int64> &out, const std::vector<cytnx_int64> &inL,
+                       const std::vector<cytnx_int64> &inR);
+    void combine_rule_(cytnx_int64 &out, const cytnx_int64 &inL, const cytnx_int64 &inR,
+                       const bool &is_reverse);
+    void reverse_rule_(cytnx_int64 &out, const cytnx_int64 &in);
+    fparity get_fparity(const cytnx_int64 &in_qnum);
```

Comment:
Should add override keywords for `get_fparity`, `is_fermionic`, and `stype_str`. Additionally, `virtual` keyword on `is_fermionic` should be removed.